src/view/observer/mutationobserver.js

```diff
@@ -244,11 +244,14 @@ export default class MutationObserver extends Observer {
 			}
 		}
 
-		this.document.fire( 'mutations', viewMutations, viewSelection );
+		// In case only non-relevant mutations were recorded it skip event / force render for the sake of performance (#5600).
```

Comment:
```suggestion
		// In case only non-relevant mutations were recorded it skips the event & force render (#5600).
```

This is only logic to not do those two things. Improved performance is a side effect.